openmdao/docs/theory_manual/solver_api.rst

```diff
@@ -0,0 +1,87 @@
+.. _theory_solver_api:
+
+****************
+OpenMDAO Solvers
+****************
+
+Every OpenMDAO solver is either a linear solver, inheriting from the :code:`LinearSolver` class, or
+a nonlinear solver, inheriting from the :code:`NonlinearSolver` class. A solver can be either monolithic
+or recursive in behavior.  Monolithic solvers typically construct some sort of linearization of
```

Comment:
"Monolithic solvers treat the associated system as a single block. Recursive solvers, as the name suggests, recurse down through their system hierarchy asking each sub-system to operate on itself. "

Reply:
docs updated